intranet/apps/announcements/models.py

```diff
@@ -167,7 +167,7 @@ def is_visible_requester(self, user):
 
     def is_visible_submitter(self, user):
         try:
-            return (self.announcementrequest and user == self.announcementrequest.user) or self.user == user
+            return (self.announcementrequest and user.id == self.announcementrequest.user_id) or self.user_id == user.id
```

Comment:
Where is `user_id` defined?

Reply:
Django implicitly generates `_id` fields for `ForeignKey`s.

Reply:
Assuming that the primary key is `id`, of course -- which it is for our `User` model.